internal/namespace/canonicalization_test.go

```diff
@@ -375,6 +375,39 @@ func TestCanonicalization(t *testing.T) {
 				"second": computedKeyPrefix + "bfc8d945d7030961",
 			},
 		},
+		{
+			"canonicalization with functioned arrow expressions",
+			ns.Namespace(
+				"document",
+				ns.MustRelation("owner", nil),
+				ns.MustRelation("viewer", nil),
+				ns.MustRelation("first", ns.Union(
+					ns.TupleToUserset("owner", "something"),
+				)),
+				ns.MustRelation("second", ns.Union(
+					ns.TupleToUserset("owner", "something"),
+				)),
+				ns.MustRelation("difftuple", ns.Union(
+					ns.TupleToUserset("viewer", "something"),
+				)),
+				ns.MustRelation("third", ns.Union(
+					ns.MustFunctionedTupleToUserset("owner", "any", "something"),
+				)),
+				ns.MustRelation("thirdwithall", ns.Union(
+					ns.MustFunctionedTupleToUserset("owner", "all", "something"),
```

Comment:
Could you add some cases in here with expressions containing the fn arrows to make sure they canonicalize the same?

i.e. `owner.all(something) + foo` == `foo + owner.all(something)`

Reply:
Added